component/grpc/observability.go

```diff
@@ -101,8 +101,11 @@ func (o *observer) log(err error) {
 	}
 	if err != nil {
 		fields["error"] = err.Error()
+		log.Sub(fields).Error()
+		return
 	}
-	log.Sub(fields).Error()
+
```

Comment:
Ok so before I changed the log level from DEBUG to ERROR, it appears that all gRPC requests were logged as DEBUG, even when there was no error. @mantzas was that intentional? It seems a bit excessive to me, usually I'd do that with a TRACE level, but it doesn't exist here. My suggestion would be not to log anything in absence of an error.

Reply:
this makes sense to me tbh, we could remove the line with Debug. Or maybe there is a reason for this that i am missing ?

Reply:
Debug is the lowest level and should be only used when debugging.
Production should run usually Info and above.

Reply:
I think we should have the possibility to see successful requests going through. Since the current lowest level is Debug, it should be printed on debug level, not completely omitted. If we do not want to see these logs, they should be hidden through configuration (using the logger level), not hardcoded.

A Trace level would work well for that, but there's a discussion to have on whether to add it or not - I feel it's not in the scope on this PR.

Reply:
Are we ok on merging this?

Reply:
Since there is not answer here I will forward and merge this since we have our approvals.

Reply:
Should we merge it?